Index: Palindromes.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>#Using stacks to check if a word is a palindrome\r\n\r\nclass Palindromes:\r\n    def _init_(self):\r\n        self.stack=[]\r\n    #Push the characters into a stack\r\n    def push(self,characters):\r\n        self.stack.append(characters)\r\n\r\n    def pop(self):\r\n        return  self.stack.pop()\r\n\r\ndef is_palindrome(s):\r\n\r\n    check=Palindromes()\r\n\r\n    for char in s:\r\n        check.push(char)\r\n\r\n    for i in range(len(s) //2):\r\n        if check.pop()!=s[i]:\r\n         return False\r\n    return True\r\n\r\n\r\ns=input(\"Enter a word:\\n\").strip()\r\n#To abide by the constraints given\r\nif not (1<=len(s)<=50):\r\n    print(\"This value is bot allowed\")\r\n    exit()\r\nif is_palindrome(s):\r\n    print(f\"The word is, {s}, is a palindrome.\")\r\n\r\nelse:\r\n    print(f\"The word is, {s}, is not palindrome.\")
===================================================================
diff --git a/Palindromes.py b/Palindromes.py
--- a/Palindromes.py	(revision 88e4ebcbe69bfd6a4f88fd14d96104e93a23e5bf)
+++ b/Palindromes.py	(date 1748254384871)
@@ -1,7 +1,9 @@
 #Using stacks to check if a word is a palindrome
+import re  #clean the input(removes anything that is not a lowercase)
 
-class Palindromes:
-    def _init_(self):
+
+class Palindrome:
+    def __init__(self):
         self.stack=[]
     #Push the characters into a stack
     def push(self,characters):
@@ -10,26 +12,27 @@
     def pop(self):
         return  self.stack.pop()
 
-def is_palindrome(s):
+def is_palindrome(word):
+    word = re.sub(r'[^a-z]', '', word.lower())  #converts to lower case and removes all non-letter characters
 
-    check=Palindromes()
+    check=Palindrome()
 
-    for char in s:
+    for char in word:
         check.push(char)
 
-    for i in range(len(s) //2):
-        if check.pop()!=s[i]:
+    for i in range(len(word) //2):
+        if check.pop()!=word[i]:
          return False
     return True
 
 
-s=input("Enter a word:\n").strip()
-#To abide by the constraints given
-if not (1<=len(s)<=50):
-    print("This value is bot allowed")
+word=input("Enter a word:\n").strip().lower() #,strip() -> ignores whitespaces
+
+if not (1<=len(word)<=50):
+    print("This value is invalid")
     exit()
-if is_palindrome(s):
-    print(f"The word is, {s}, is a palindrome.")
+if is_palindrome(word):
+    print(f"The word {word}, is a palindrome.")
 
 else:
-    print(f"The word is, {s}, is not palindrome.")
\ No newline at end of file
+    print(f"The word {word}, is not palindrome.")
\ No newline at end of file
